crates/starknet-devnet/tests/test_simulate_transactions.rs

```diff
@@ -232,6 +234,204 @@ mod simulation_tests {
         );
     }
 
+    async fn update_gas_scenario(devnet: BackgroundDevnet, assert_chain_id: &str) {
```

Comment:
Either this function should have a more descriptive name or it should have triple-slash documentation above.

Reply:
chain id might be wrong since we need to reinitialized `Starknet::init_block_context` so this is why I added this extra check based on my knowledge about implementation.

Reply:
Added comments in 4f55b2ec9f38a6aadf98b04e5a669454f3a98a73